tests/pytests/test_vecsim.py

```diff
@@ -1357,3 +1356,45 @@ def test_rdb_memory_limit():
 
     # reset env (for clean RLTest run with env reuse)
     env.assertTrue(conn.execute_command('CONFIG SET', 'maxmemory', '0'))
+
+def test_timeout_reached():
+    env = Env(moduleArgs='DEFAULT_DIALECT 2')
+    conn = getConnectionByEnv(env)
+    nshards = env.shardsCount
+
+    vecsim_algorithms_and_sizes = [('FLAT', 80000 * nshards), ('HNSW', 10000 * nshards)]
+    hybrid_modes = ['BATCHES', 'ADHOC_BF']
+    dim = 10
+
+    for algo, n_vec in vecsim_algorithms_and_sizes:
+        # succeed to create indexes with no limits
+        query_vec = load_vectors_to_redis(env, n_vec, 0, dim)
+        env.expect('FT.CREATE', 'idx', 'SCHEMA', 'vector', 'VECTOR', algo, '8', 'TYPE', 'FLOAT32',
+                   'DIM', dim, 'DISTANCE_METRIC', 'L2', 'INITIAL_CAP', n_vec).ok()
+        waitForIndex(env, 'idx')
+
+        # STANDART KNN
+        # run query with no timeout. should succeed.
+        res = conn.execute_command('FT.SEARCH', 'idx', '*=>[KNN $K @vector $vec_param]', 'NOCONTENT', 'LIMIT', 0, n_vec,
+                                   'PARAMS', 4, 'K', n_vec, 'vec_param', query_vec.tobytes(),
+                                   'TIMEOUT', 0)
+        env.assertEqual(res[0], n_vec)
+        # run query with 1 milisecond timeout. should fail.
+        res = conn.execute_command('FT.SEARCH', 'idx', '*=>[KNN $K @vector $vec_param]', 'NOCONTENT', 'LIMIT', 0, n_vec,
+                                   'PARAMS', 4, 'K', n_vec, 'vec_param', query_vec.tobytes(),
+                                   'TIMEOUT', 1)
+        env.assertEqual(res[0], 0)
+
+        # HYBRID MODES
+        for mode in hybrid_modes:
+            res = conn.execute_command('FT.SEARCH', 'idx', '(-dummy)=>[KNN $K @vector $vec_param HYBRID_POLICY $hp]', 'NOCONTENT', 'LIMIT', 0, n_vec,
+                                       'PARAMS', 6, 'K', n_vec, 'vec_param', query_vec.tobytes(), 'hp', mode,
+                                       'TIMEOUT', 0)
+            env.assertEqual(res[0], n_vec)
+
+            res = conn.execute_command('FT.SEARCH', 'idx', '(-dummy)=>[KNN $K @vector $vec_param HYBRID_POLICY $hp]', 'NOCONTENT', 'LIMIT', 0, n_vec,
+                                       'PARAMS', 6, 'K', n_vec, 'vec_param', query_vec.tobytes(), 'hp', mode,
+                                       'TIMEOUT', 1)
+            env.assertEqual(res[0], 0)
```

Comment:
Consider verifying that TIMEOUT has been caught (by setting the mode in which TIMEOUT is considered as error)